tools/prerelease-tool/main.go

```diff
@@ -0,0 +1,134 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"sort"
+	"time"
+
+	"github.com/google/go-github/github"
+	"golang.org/x/oauth2"
+)
+
+type prereleaseTool struct {
+	Data        []*github.RepositoryRelease
+	Limit       int
+	Owner       string
+	Repo        string
+	client      *github.Client
+	Prereleases []github.RepositoryRelease
+	DryRun      bool
+}
+
+func main() {
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+	token, ok := os.LookupEnv("GITHUB_TOKEN")
+	if !ok {
+		log.Fatal("Github token must be provided through GITHUB_TOKEN environment variable")
+	}
+	ts := oauth2.StaticTokenSource(
+		&oauth2.Token{AccessToken: token},
+	)
+	tc := oauth2.NewClient(ctx, ts)
+	client := github.NewClient(tc)
+	dryRun := flag.Bool("dry-run", false, "perform a dry run instead of executing create/delete/update actions")
+	limit := flag.Int("limit", 7, "limit of prereleases to keep")
+	owner := flag.String("owner", "filecoin-project", "github owner or organization")
+	repo := flag.String("repo", "go-filecoin", "github project repository")
+	flag.Parse()
+	r := prereleaseTool{
+		Limit:  *limit,
+		Owner:  *owner,
+		Repo:   *repo,
+		client: client,
+		DryRun: *dryRun,
+	}
+	if err := r.get(ctx); err != nil {
+		log.Fatalf("Could not find any releases: %+v", err)
+	}
+	r.getPrereleases()
+	r.sortReleasesByDate()
+	ok, err := r.trim(ctx, r.oldReleaseIDs())
+	if err != nil {
+		log.Fatalf("Problem attempting to delete releases: %+v", err)
+	}
+	if !ok {
+		log.Print("Remove --dry-run flag to apply changes")
+	}
+}
+
+func (r *prereleaseTool) get(ctx context.Context) error {
```

Comment:
This gets all releases, correct? This means things outside of just the nightly's? If so can we call it `getReleases()`